copy ssh key to administrator authorized keys

iankingori opened the pull request:
Admin Users require to have the public key added to c:/ProgramData/ssh/administrators_authorized_keys. [reference](https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_keymanagement#administrative-user)

jsturtevant commented:
/ok-to-test

jsturtevant commented:
I wonder if the ssh issue is the same as the annual channel: https://github.com/kubernetes-sigs/windows-testing/pull/423/files#diff-327b21e02c78e72822fcacc34f880ce04e20d5b131f2a91f4901849392afe897

iankingori commented:
/retest

iankingori commented:
/retest

iankingori commented:
/retest

iankingori commented:
> I wonder if the ssh issue is the same as the annual channel: https://github.com/kubernetes-sigs/windows-testing/pull/423/files#diff-327b21e02c78e72822fcacc34f880ce04e20d5b131f2a91f4901849392afe897

ssh seems to be working just auth is failing: `Failed to get logs for Machine capz-conf-pwn65v-md-win-tsqf6-qrwxx, Cluster default/capz-conf-pwn65v: getting a new SSH client connection: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain`. Was this the same issue with annual channel?

jsturtevant commented:
Was using the new Github ui and it seems it doesn't let me trigger the "approve and run" so the linters and other checks weren't run.  It look like `capz / capz-templates (pull_request) ` failed since the  `make generate-templates` wasn't run.  The way to edit these files is to make the change to `windows-base` (or any of the patch files), then run `make generate-templates` which will update the rest of the files appropriately. In this case since we want all the templates to have the same configuration, we can just edit the `windows-base.yaml`

jsturtevant commented:
I've also added https://github.com/kubernetes/test-infra/pull/34136 so we can see if this fixes WS2025

iankingori commented:
> Was using the new Github ui and it seems it doesn't let me trigger the "approve and run" so the linters and other checks weren't run. It look like `capz / capz-templates (pull_request) ` failed since the `make generate-templates` wasn't run. The way to edit these files is to make the change to `windows-base` (or any of the patch files), then run `make generate-templates` which will update the rest of the files appropriately. In this case since we want all the templates to have the same configuration, we can just edit the `windows-base.yaml`

I actually ran make then figured we may not need it for the `shared-image-gallery`. I've brought the change back in

iankingori commented:
/retest

marosset commented on capz/templates/gmsa-ci.yaml:
```diff
@@ -83,6 +83,11 @@ spec:
+          ${AZURE_SSH_PUBLIC_KEY:=""}
+        owner: root:root
+        path: $env:ProgramData\ssh\administrators_authorized_keys
+        permissions: "0644"
```
nit: the `icacls.exe` in the docs linked in the PR description only grant permissions to admin and system accounts, it doesn't grant read access to users. Should we update this to 0640?

iankingori replied:
Good catch, let me dig into it

iankingori replied:
Tested and 0640 also works and pushed the change. It doesn't seem like the permissions have any impact on the file though, I'll explore further as I dig into the test failures

iankingori commented:
@jsturtevant the last run has [collected logs](https://gcsweb.k8s.io/gcs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_windows-testing/474/pull-e2e-capz-windows-2025-extension/1879147853192892416/artifacts/clusters/capz-conf-yk9rqr/machines/capz-conf-yk9rqr-md-win-5kf9n-8mtm7/) on the Windows nodes. I've made changes to the path and removed `$env:ProgramData` and replaced it with `C:\ProgramData`

jsturtevant commented:
/lgtm
/approve
I am going to merge this PR so we can start collecting logs on WS2025 to diagnose the failures further.
